### DISCUSSION TITLE
How to orderBy a colonne by ENUM type ?

### DISCUSSION THREAD

**Discussion by foxyrs** [simolus3/drift#1420] [Q&A]
Hello

I'm trying to sort my column "status" by ENUM type.

I tried to create and add a customExpression to my Moor's query (like in the script below).

But it didn't work.

I would know if it is possible to order a column by String without using customSelect().

```
CustomExpression customOrder = CustomExpression(
          "ORDER BY FIELD (status, 'INIT','TAKING_CARE','WAITING_EXPERT','WAITING_RESULT','OK','CLOSE','ERROR','OTHER')");
      query.orderBy([OrderingTerm(expression: customOrder)]);
```

Thanks

**Comment by simolus3** [comment]
You can't use a custom expression here because ordering clauses are no expressions. Further, sqlite3 does not support the `FIELD` syntax.

One option would be to map each enum to an int with [`caseMatch`](https://611a8b35e1c162009a7aca08--moor.netlify.app/api/moor/expression/casematch):

```dart
query.orderBy(OrderingTerm.asc(
  yourTable.status.caseMatch<int>(
    when: {
      for (var i = 0; i < YourEnum.values.length; i++)
        Constant(YourEnum.values[i].toString()): Constant(i)
    }
  )
))
```

This will generate something like `CASE status WHEN 'INIT' THEN 1 WHEN 'TAKING_CARE' THEN 2 ... END`, which isn't particular efficient I imagine.

**Reply by foxyrs** [reply]
Tanks for your answer, I understand my previous mistake.

I'm using Moor V 3.4.0 and caseMatch isn't available. 
Does exist an alternative for this version ?

Actually i'm blocked with Dart 2.12.3 due to the other libraries on my project.

**Reply by simolus3** [reply]
>  Does exist an alternative for this version ?

You could take a look at [the implementation](https://github.com/simolus3/moor/blob/29453812133ff1f13e6dc1637addddf4bc641ebc/moor/lib/src/runtime/query_builder/expressions/expression.dart#L116-L125), including the one of [case when](https://github.com/simolus3/moor/blob/29453812133ff1f13e6dc1637addddf4bc641ebc/moor/lib/src/runtime/query_builder/expressions/case_when.dart). Probably you can just copy that.

> Actually i'm blocked with Dart 2.12.3 due to the other libraries on my project.

Do they really not support Dart 2.13? What goes wrong if you use the latest version?
